java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -265,33 +408,78 @@ public HostColumnVector copyToHost() {
       HostMemoryBuffer hostDataBuffer = null;
       HostMemoryBuffer hostValidityBuffer = null;
       HostMemoryBuffer hostOffsetsBuffer = null;
-      BaseDeviceMemoryBuffer valid = offHeap.getValid();
-      BaseDeviceMemoryBuffer offsets = offHeap.getOffsets();
-      BaseDeviceMemoryBuffer data = offHeap.getData();
+      BaseDeviceMemoryBuffer valid = (BaseDeviceMemoryBuffer)getValidityBuffer();
```

Comment:
Another nit for follow on work. After refactoring the types being returned we probably could combine the majority of this code with the code from `copyToHostNestedHelper` so that all this does is call into `copyToHostNestedHelper` and then has an API to translate the top level `HostColumnVector.NestedHostColumnVector` into a `HostColumnVector`.  But again lets get this in as is and worry about that later.

Reply:
Templatized the interface to get rid of the casting.

Reply:
Sorry wrong reply here. I will do this as follow on.

Reply:
Leaving this comment so we can capture the follow on work.

Reply:
I did add the needed to change to take this out.